File: Exercism Codes/Python/pig_latin.py
def translate(text):
    result = []

    for c in text.split(" "):
        sl = 0
        while True:
            is_vowel = c[sl] in "aeiou"
            rule_1 = c[sl:sl + 2] == "yt" or c[sl:sl + 2] == "xr"
            rule_4 = c[sl - 1:sl] not in "aeiou" and c[sl] == "y"
            if is_vowel or rule_1 or rule_4:
                break

            if c[sl:sl + 2] == "qu":
                sl += 1

            sl += 1

        result.append(c[sl:] + c[:sl] + "ay")

    return " ".join(result)

File: Exercism Codes/JavaScript/pig-latin.js
export const translate = (text) => {
  const result = [];

  for (const x of text.split(" ")) {
    let sl = 0;
    while (true) {
      const isVowel = "aeiou".includes(x[sl]);
      const rule1 = x.substring(sl, sl + 2) === "yt" || x.substring(sl, sl + 2) === "xr";
      const rule4 = (!("aeiou".includes(x[sl - 1])) && x[sl - 1] !== undefined) && x[sl] === "y";

      if (isVowel || rule1 || rule4) {
        break;
      }

      if (x.substring(sl, sl + 2) === "qu") {
        sl += 1;
      }

      sl += 1;
    }
    result.push(x.substring(sl) + x.substring(0, sl) + "ay");
  }

  return result.join(" ");
};